src/execution_plan/ops/op.h

```diff
@@ -96,5 +97,13 @@ Record OpBase_Consume(OpBase *op);  // Consume op.
 Record OpBase_Profile(OpBase *op);  // Profile op.
 int OpBase_ToString(const OpBase *op, char *buff, uint buff_len);
 
+/* If an operation holds the sole reference to a Record it is evaluating,
+ * that reference should be tracked so that it may be freed in the event of a run-time error. */
+void OpBase_AddVolatileRecord(OpBase *op, const Record r);
```

Comment:
If we switch to a single record, we can auto set it in the consume call, we'll only have to mange its unset (when ever we're done with that record)